Core/ELF/ParamSFO.cpp

```diff
@@ -276,3 +280,34 @@ void ParamSFOData::ValueData::SetData(const u8* data, int size)
 	u_size = size;
 }
 
+std::string ParamSFOData::generateFakeID(std::string filename)
+{
+	// Generates fake gameID for homebrew based on it's folder name.
+	// Should probably not be a part of ParamSFO, but it'll be called in same places.
+	std::string file = PSP_CoreParameter().fileToStart;
+	if (filename != "")
+		file = filename;
+
+	std::size_t lslash = file.find_last_of("/");
+	file = file.substr(lslash + 1);
+
+	int letter, sumOfAllLetters = 0;
+	for (int i = 0; i < file.size(); i++) {
+		letter = file[i];
+		sumOfAllLetters += letter;
+	}
+	
+	size_t j;
+	for (j = 0; j < file.size(); j++) {
+		file[j] = toupper(file[j]);
+	}
+	if (file.size() < 4) {
+		file += "HOME";
+	}
+	file = file.substr(0, 4);
+
+	std::stringstream s;
+	s << std::setfill('0') << std::setw(5) << StringFromFormat("%i", sumOfAllLetters);
```

Comment:
I'm not sure this works, and even if it does, why not just use the format directly?

```c++
std::string fakeID = file + StringFromFormat("%05i", sumOfAllLetters);
```

-[unknown]